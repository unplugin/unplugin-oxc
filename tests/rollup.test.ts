import path from 'node:path'
import { rollupBuild, testFixtures } from '@sxzz/test-utils'
import { describe } from 'vitest'
import UnpluginOxc from '../src/rollup'

describe('rollup', async () => {
  const { dirname } = import.meta
  await testFixtures(
    '*.ts',
    async (args, id) => {
      const { snapshot } = await rollupBuild(id, [
        UnpluginOxc({
          transform: {
            target: 'es2015',
          },
        }),
      ])
      return snapshot
    },
    { cwd: path.resolve(dirname, 'fixtures'), promise: true },
  )
})
