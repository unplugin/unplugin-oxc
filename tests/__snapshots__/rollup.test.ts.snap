// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rollup > basic.ts 1`] = `
"// basic.js
import { jsx } from 'react/jsx-runtime';

const node = jsx("div", {});

const num = 10;
let Color = function(Color) {
	Color[Color["Red"] = 1] = "Red";
	Color[Color["Green"] = 2] = "Green";
	Color[Color["Blue"] = 3] = "Blue";
	return Color;
}({});
class Test {
	constructor(field = 10) {
		this.field = field;
	}
}
let NS;
(function(_NS) {
	const foo = _NS.foo = 1;
})(NS || (NS = {}));
let optionalChain = NS === null || NS === void 0 ? void 0 : NS.foo;

export { Color, node, num, optionalChain };
"
`;
